### ISSUE TITLE
Implement pull from insights-content-service and load groups configuration into memory

### ISSUE THREAD

**Issue by joselsegura** [RedHatInsights/insights-results-smart-proxy#18]
TL;DR
As explained in issue #7, the same idea should be implemented for groups configuration

The insights-content-service is already serving the groups configuration. The smart proxy should retrieve that configuration and update it periodically and store it in data structures in memory to being able to respond to group requests quickly

### Acceptance criteria

- Smart proxy is able to respond to `/groups` endpoint
- Smart proxy retrieves groups configuration from the `content-service` at start up
- Smart proxy retrieves periodically the groups from `content-service`, acting as cache when the service is not available

### Unknowns
- Which transport to use to avoid marshaling and unmarshaling same structures in both services
